### PR TITLE
Merge Cosmos.Sql provider to current release

### PR DESCRIPTION
**Pull request by smitpatel**

### CODE REVIEW

**Comment by smitpatel** [comment]
@dotnet-bot test OSX 10.12 release build

File: test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs [hunk 1]
```diff
@@ -15,6 +15,8 @@ public class NorthwindQueryCosmosSqlFixture<TModelCustomizer> : NorthwindQueryFi
     {
         protected override ITestStoreFactory TestStoreFactory => CosmosSqlNorthwindTestStoreFactory.Instance;
 
+        protected override bool UsePooling => false;
```
Comment (AndriySvyryd) [inline test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs]: Why no pooling?

Comment (smitpatel) [reply test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs]: Something failed in Reset method.

File: test/EFCore.Cosmos.Sql.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            get
+            {
+                var documentClient = new DocumentClient(
+                    new Uri(TestEnvironment.DefaultConnection),
+                    TestEnvironment.AuthToken);
+
+                try
+                {
+                    documentClient.OpenAsync().GetAwaiter().GetResult();
```
Comment (AndriySvyryd) [inline test/EFCore.Cosmos.Sql.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs]: Make sure this is only executed once per run

File: test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs (hunk 1 above)
Comment (AndriySvyryd) [reply test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs]: File an issue?

Comment (smitpatel) [reply test/EFCore.Cosmos.Sql.FunctionalTests/Query/NorthwindQueryCosmosSqlFixture.cs]: #12878

**Pull request merged**